diff --git a/libs/vkd3d/device.c b/libs/vkd3d/device.c
index 4a25216..b84bbf9 100644
--- a/libs/vkd3d/device.c
+++ b/libs/vkd3d/device.c
@@ -10774,10 +10774,13 @@ bool d3d12_device_validate_shader_meta(struct d3d12_device *device, const struct
 
             if (!allowed_edge_case)
             {
-                ERR("Required WaveSize range [%u, %u], but supported range is [%u, %u].\n",
-                        meta->cs_wave_size_min, meta->cs_wave_size_max,
+                ERR("Shader %016"PRIx64" requires WaveSize range [%u, %u], but supported range is [%u, %u]; "
+                        "flags %#x, workgroup [%u, %u, %u], preferred %u.\n",
+                        meta->hash, meta->cs_wave_size_min, meta->cs_wave_size_max,
                         device->d3d12_caps.options1.WaveLaneCountMin,
-                        device->d3d12_caps.options1.WaveLaneCountMax);
+                        device->d3d12_caps.options1.WaveLaneCountMax, meta->flags,
+                        meta->cs_workgroup_size[0], meta->cs_workgroup_size[1],
+                        meta->cs_workgroup_size[2], meta->cs_wave_size_preferred);
                 return false;
             }
         }
